curvefs/src/mds/mds_service.cpp

```diff
@@ -0,0 +1,226 @@
+/*
+ *  Copyright (c) 2021 NetEase Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License") {
+    return;
+}
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+/*
+ * Project: curve
+ * Created Date: 2021-05-19
+ * Author: chenwei
+ */
+#include "curvefs/src/mds/mds_service.h"
+
+namespace curvefs {
+namespace mds {
+void MdsServiceImpl::CreateFs(::google::protobuf::RpcController* controller,
+                    const ::curvefs::mds::CreateFsRequest* request,
+                    ::curvefs::mds::CreateFsResponse* response,
+                    ::google::protobuf::Closure* done) {
+    brpc::ClosureGuard doneGuard(done);
+    brpc::Controller* cntl = static_cast<brpc::Controller*>(controller);
+    std::string fsName = request->fsname();
+    uint64_t blockSize = request->blocksize();
+    FSType type = request->fstype();
+    if (type == FSType::TYPE_VOLUME) {
+        if (!request->has_volume()) {
+            response->set_statuscode(FSStatusCode::PARAM_ERROR);
+            LOG(ERROR) << "CreateFs request, type is volume, but has no volume"
+                       << ", fsName = " << fsName;
+            return;
+        }
+        curvefs::common::Volume volume = request->volume();
+        LOG(INFO) << "CreateFs request, fsName = " << fsName
+                << ", blockSize = " << blockSize
+                << ", volume.volumeName = " << volume.volumename();
+        FSStatusCode status = fsManager_->CreateFs(fsName, blockSize,
```

Comment:
Concurrency issues need to be considered.

Reply:
Consider in the next version.